### PR TITLE
fix: Fix title generation for Anthropic agents behind reverse proxy

### PR DESCRIPTION
**Pull request by mawburn**
## Summary

When using Anthropic agents with a reverse proxy, title generation was failing with 404 errors. The problem was that the agent configuration wasn't properly passing the custom API URL to the client that handles title generation.

This fix moves the provider-specific configuration (like custom API URLs and keys) from the run creation logic into the AgentClient itself, so all agent operations, including title generation, use the correct endpoints. This ensures that agents behind reverse proxies work correctly for all operations, not just the main chat functionality.

## Testing

Create an Agent using the Anthropic endpoint behind an `ANTHROPIC_REVERSE_PROXY` and without this change, the `gen_title` response should consistently return a 404. 

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes

## Copilot Summary

This pull request introduces enhancements and fixes across multiple files, primarily focusing on testing frameworks, configuration handling, and provider-specific logic for agents and endpoints. The changes aim to improve test coverage, streamline configuration inheritance, and address edge cases for various providers.

### Enhancements to testing frameworks:
* [`api/server/controllers/agents/__tests__/run.spec.js`](diffhunk://#diff-12f2c0bf2bfbc5c39063cf52a4ba35ebe79ba0d841284c11af6ddc923972b1b9R1-R290): Added comprehensive unit tests for the `createRun` function, covering scenarios such as provider endpoint mapping, reasoning key configuration, and stream buffer handling. Mocking libraries and utilities were introduced to simulate various agent configurations and provider behaviors.
* [`api/server/services/Endpoints/anthropic/__tests__/llm.spec.js`](diffhunk://#diff-95cfd26034f21f24c5328a1951218b92a5dbcc05a794d489edb25f83d358029eR1-R367): Implemented detailed tests for `getLLMConfig`, including handling of system options, model-specific parameters, proxy configurations, and edge cases. Mocked dependencies ensure robust validation of configuration logic.

### Configuration handling improvements:
* [`api/server/controllers/agents/client.js`](diffhunk://#diff-8c99b1ddca0d7fa6758088ffec41bed3ff6ce185163505281c9e3de03450e8aaL651-R651): Updated `AgentClient` to inherit provider-specific configurations, such as `clientOptions`, `apiKey`, and `anthropicApiUrl`, from the agent's `model_parameters`. Added a default recursion limit of `25` to `agentsEConfig`. [[1]](diffhunk://#diff-8c99b1ddca0d7fa6758088ffec41bed3ff6ce185163505281c9e3de03450e8aaL651-R651) [[2]](diffhunk://#diff-8c99b1ddca0d7fa6758088ffec41bed3ff6ce185163505281c9e3de03450e8aaR932-R943)
* [`api/server/services/Endpoints/anthropic/llm.js`](diffhunk://#diff-fc481ba7bec2b8c192a1f98b3051afa0db36c0b1057c3bec8fbd0b100c7c618dR75): Enhanced `getLLMConfig` to include `anthropicApiUrl` when a `reverseProxyUrl` is provided, ensuring consistent proxy handling.

### CODE REVIEW

File: api/server/controllers/agents/run.js [hunk 1]
```diff
@@ -50,6 +50,25 @@ async function createRun({
...
+    if (ANTHROPIC_API_KEY && ANTHROPIC_API_KEY !== 'user_provided' && !llmConfig.apiKey) {
+      llmConfig.apiKey = ANTHROPIC_API_KEY;
+    }
+
+    if (ANTHROPIC_REVERSE_PROXY) {
+      if (!llmConfig.clientOptions) {
+        llmConfig.clientOptions = {};
+      }
+
+      const baseURL = ANTHROPIC_REVERSE_PROXY.replace(/\/v1\/?$/, '');
```
Comment (nhtruong) [inline api/server/controllers/agents/run.js]: Why remove `/v1` specifically? is this a common pattern in reverse proxy URL?

Comment (mawburn) [reply api/server/controllers/agents/run.js]: Yeah, good point. This might be relevant to just our proxy.

**Comment by danny-avila** [comment]
Just adding to discussion, the `llmConfig` is defined here:
https://github.com/danny-avila/LibreChat/blob/main/api/server/services/Endpoints/anthropic/initialize.js

via `getLLMConfig`:
https://github.com/danny-avila/LibreChat/blob/main/api/server/services/Endpoints/anthropic/llm.js

if the title method is missing something, I think the current `titleConvo` method would need to change before any change to `createRun` call.

File: api/server/controllers/agents/run.js (hunk 1 above)
Comment (mawburn) [reply api/server/controllers/agents/run.js]: Going a different direction.

File: api/server/controllers/agents/client.js [hunk 2]
```diff
@@ -648,7 +648,7 @@ class AgentClient extends BaseClient {
           user_id: this.user ?? this.options.req.user?.id,
           hide_sequential_outputs: this.options.agent.hide_sequential_outputs,
         },
-        recursionLimit: agentsEConfig?.recursionLimit,
+        recursionLimit: agentsEConfig?.recursionLimit ?? 25,
```
Comment (mawburn) [inline api/server/controllers/agents/client.js:675]: This was causing tests to fail because it was undefined:
https://github.com/danny-avila/LibreChat/actions/runs/15478430382/job/43579331938

So, I updated it to default to what the docs [say](https://www.librechat.ai/docs/features/agents#agent-chain:~:text=If%20left%20unconfigured%2C%20the%20default%20is%2025%20steps%2C%20but%20you%20can%20adjust%20this%20to%20suit%20your%20needs.%20For%20admins%2C%20you%20can%20set%20a%20global%20default%20as%20well%20as%20a%20global%20maximum%20in%20the%20librechat.yaml%20file.).
